docs/backend/deprecation.md

```diff
@@ -0,0 +1,311 @@
+
+> DeprecationWarning: old.pkg.foo has moved to new.pkg.bar.
+> Import of old.pkg.foo will become unsupported in Version 2.0
+
+### Moving Whole Packages
+
+This is the same as moving a module, just create for each module a file.
+
+### Deprecating methods and properties
+
+You can use the `@deprecate` decorator from [zope.deprecation Deprecating methods and properties](http://docs.zope.org/zope.deprecation/api.html#deprecating-methods-and-properties) to deprecate methods in a module:
+
+```python
+from zope.deprecation import deprecate
+
+@deprecate('Old method is no longer supported, use new_method instead.')
+def old_method():
+    return 'some value'
+```
+
+The `deprecated` wrapper method is for deprecating properties:
+
+```python
+from zope.deprecation import deprecated
+
+foo = None
+foo = deprecated(foo, 'foo is no more, use bar instead')
+```
+
+### Moving functions and classes
+
+Given we have a Python file at `old/foo/bar.py` and want to move some classes or functions to `new/baz/baaz.py`.
+
+Here `zope.deferredimport` offers a deprecation helper.
+It also avoids circular imports on initialization time.
+
+```python
+import zope.deferredimport
+zope.deferredimport.initialize()
+
+zope.deferredimport.deprecated(
+    "Import from new.baz.baaz instead",
+    SomeOldClass='new.baz:baaz.SomeMovedClass',
+    some_old_function='new.baz:baaz.some_moved_function',
+)
+
+def some_function_which_is_not_touched_at_all():
+    pass
+```
+
+### Deprecating a GenericSetup profile
+
+Starting with GenericSetup 1.8.2 (part of Plone > 5.0.2) the `post_handler` attribute in ZCML can be used to call a function after the profile was applied.
+We use this feature to issue a warning.
+
+First we register the same profile twice. Under the new name and under the old name:
+
+```xml
+<genericsetup:registerProfile
+    name="default"
+    title="My Fance Package"
+    directory="profiles/default"
+    description="..."
+    provides="Products.GenericSetup.interfaces.EXTENSION"
+    />
+
+<genericsetup:registerProfile
+    name="some_confusing_name"
+    title="My Fance Package (deprecated)"
+    directory="profiles/some_confusing_name"
+    description="... (use profile default instaed)"
+    provides="Products.GenericSetup.interfaces.EXTENSION"
+    post_handler=".setuphandlers.deprecate_profile_some_confusing_name"
+    />
+```
+
+And in `setuphandlers.py` add a function:
+
+```python
+import warnings
+
+def deprecate_profile_some_confusing_name(tool):
+    warnings.warn(
+        'The profile with id "some_confusing_name" was renamed to "default".',
+        DeprecationWarning
+    )
+```
+
+### Deprecating a template position
+
+Sometimes we need to move templates to new locations. Since addons often use jbot to override templates by their position, we need to point them to the new position as well as make sure that the override still works with the old position.
```

Comment:
```suggestion
Sometimes we need to move templates to new locations. Since addons often use [z3c.jbot](https://github.com/zopefoundation/z3c.jbot) to override templates by their position, we need to point them to the new position as well as make sure that the override still works with the old position.
```